src/OutputWriters/output_construction.jl

```diff
@@ -46,6 +46,38 @@ function construct_output(user_output::Union{AbstractField, Reduction}, grid, us
     return construct_output(user_output, indices)
 end
 
+
+const WindowedData = OffsetArray{<:Any, <:Any, <:SubArray}
+const WindowedField = Field{<:Any, <:Any, <:Any, <:Any, <:Any, <:Any, <:WindowedData}
+const ComputedField = Field{<:Any, <:Any, <:Any, <:AbstractOperation}
+
+function construct_output(user_output::WindowedField, grid, user_indices, with_halos)
+    if !with_halos
+        new_indices = restrict_to_interior.(user_output.indices, location(user_output), topology(grid), size(grid))
+        return view(user_output, new_indices...)
+    end
+
+    return user_output
+end
+
+function construct_output(user_output::ComputedField, grid, user_indices, with_halos)
+    if indices(user_output) == (Colon(), Colon(), Colon())
+        return construct_output(user_output.operand, grid, user_indices, with_halos)
+    else
+        @info "Only change halos"
+        if !with_halos
+            new_indices = [ indexes == Colon() ? restrict_to_interior(indexes, loc, topo, grid_size) : indexes
+                           for (indexes, loc, topo, grid_size) in zip(indices(user_output), location(user_output), topology(grid), size(grid)) ]
+            @show new_indices summary(user_output.data)
+            return view(user_output, new_indices...)
```

Comment:
Ooh, that makes sense!

In that case, perhaps an if-statement that checks if the output _already_ has non default indices is simplest? There might be a little bit of logic to do to figure out the right way to express it so it's generally valid whether user output is a field (which has indices attached) or an AbstractOperation (which does not)

Reply:
I don't think you'll be able to convert ComputedField to have SubArray underneath without doing something non-logical, like allocating more memory than we need to represent the computation. Instead I guess we should recognize that the information we need is not contained in the type of .data. Instead, it's contained in the type (and value) of indices(field)

Reply:
Is it possible to rework how calculations are done so that we never have to access the underlying `Array` or `SubArray`? Meaning we would always use the `OffsetArrays` and offset indices? It seems like it would simplify a fair amount of code.

Reply:
What would it simplify?

We have `SubArray` data so that we can support `Field`s which are views into other `Field`. We then use this abstraction to support flexible output specification (at least in principle... there's still some kinks which is why this PR exists) as well as `FieldTimeSeries` built from sliced fields (esp for the common case where halos have been sliced).